src/main/java/build/buf/protovalidate/internal/evaluator/ConstraintResolver.java

```diff
@@ -32,18 +34,24 @@ class ConstraintResolver {
    * @param desc the message descriptor.
    * @return the resolved {@link MessageConstraints}.
    */
-  MessageConstraints resolveMessageConstraints(Descriptor desc) {
+  MessageConstraints resolveMessageConstraints(Descriptor desc)
+      throws InvalidProtocolBufferException {
     DescriptorProtos.MessageOptions options = desc.getOptions();
     if (!options.hasExtension(ValidateProto.message)) {
       return MessageConstraints.getDefaultInstance();
     }
-
-    MessageConstraints constraints = options.getExtension(ValidateProto.message);
-    boolean disabled = constraints.getDisabled();
-    if (disabled) {
-      return MessageConstraints.newBuilder().setDisabled(true).build();
+    // Don't use getExtension here to avoid exception if descriptor types don't match.
+    // This can occur if the extension is generated to a different Java package.
+    Object value = options.getField(ValidateProto.message.getDescriptor());
+    if (value instanceof MessageConstraints) {
+      return ((MessageConstraints) value);
+    }
+    if (value instanceof MessageLite) {
+      // Possible that this represents the same constraint type, just generated to a different
+      // java_package.
+      return MessageConstraints.parseFrom(((MessageLite) value).toByteString());
     }
-    return constraints;
+    return MessageConstraints.getDefaultInstance();
```

Comment:
Seems like this should never happen, and thus maybe be an exception, instead of silently dropping constraints but using the default instance.

Reply:
Good call - this branch was unreachable in the tests and better to throw an exception to catch sooner than transparently fail to perform validation.